src/main/application-update/install-application-update-tray-item.injectable.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import { computed } from "mobx";
+import { trayMenuItemInjectionToken } from "../tray/tray-menu-item/tray-menu-item-injection-token";
+import quitAndInstallUpdateInjectable from "../electron-app/features/quit-and-install-update.injectable";
+import discoveredUpdateVersionInjectable from "../../common/application-update/discovered-update-version/discovered-update-version.injectable";
+import updateIsBeingDownloadedInjectable from "../../common/application-update/update-is-being-downloaded/update-is-being-downloaded.injectable";
+import { withErrorSuppression } from "../../common/utils/with-error-suppression/with-error-suppression";
+import { pipeline } from "@ogre-tools/fp";
+import withErrorLoggingInjectable from "../../common/utils/with-error-logging/with-error-logging.injectable";
+
+const installApplicationUpdateTrayItemInjectable = getInjectable({
+  id: "install-update-tray-item",
+
+  instantiate: (di) => {
+    const quitAndInstallUpdate = di.inject(quitAndInstallUpdateInjectable);
+    const discoveredVersionState = di.inject(discoveredUpdateVersionInjectable);
+    const downloadingUpdateState = di.inject(updateIsBeingDownloadedInjectable);
+    const withErrorLoggingFor = di.inject(withErrorLoggingInjectable);
+
+    return {
+      id: "install-update",
+      parentId: null,
+      orderNumber: 50,
+
+      label: computed(() => {
+        const versionToBeInstalled = discoveredVersionState.value.get()?.version;
+
+        return `Install update ${versionToBeInstalled}`;
```

Comment:
This would read terribly when `.get()` returns `undefined`.

Reply:
This is the case where strict mode hurts us again. By looking code, you can immediately see that there is no such case where it would be called with `discoveredVersionState.value.get()` being `undefined`

Reply:
How is it strict modes' fault? The other option would be accidentally gettting a `TypeError` at runtime if we dereferences it in the wrong order. I would say that this shows that the design is wrong.

Reply:
Because by looking at the code, you know exactly that this function is not called without the value.

Reply:
Not here you cannot, you have to go looking at the users of this code.

Reply:
@Iku-turso Please don't just resolve conversations (unless you are using resolution as a means of marking a TODO item as complete before pushing of course).

Reply:
Ah